tools/NuGetPackager/PackCommand.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.ProjectModel;
+using Microsoft.DotNet.ProjectModel.Graph;
+using Microsoft.DotNet.ProjectModel.Utilities;
+using NuGet.Frameworks;
+using NuGet.Versioning;
+
+namespace NuGetPackager
+{
+    internal class PackCommand
+    {
+        private readonly string _baseDir;
+
+        public PackCommand(string baseDir)
+        {
+            _baseDir = baseDir;
+        }
+
+        public async Task PackAsync(string nuspec, string config, string outputDir)
+        {
+            var project = ProjectContext.Create(Path.GetDirectoryName(nuspec), FrameworkConstants.CommonFrameworks.NetCoreApp10);
+            var props = "configuration=" + config;
+            var idx = 0;
+            foreach (var depVersion in GetDependencies(project).OrderBy(p => p.Item1).Select(p => p.Item2))
+            {
+                idx++;
+                props += ";dep_" + idx + "=" + depVersion;
+            }
+
+            if (Command.CreateDotNet("build", new[] { project.ProjectFile.ProjectFilePath, "--configuration", config }, configuration: config).Execute().ExitCode != 0)
+            {
+                throw new GracefulException("Build failed");
+            }
+
+            Directory.CreateDirectory(outputDir);
+
+            var version = project.ProjectFile.Version.ToNormalizedString();
+            await Nuget("pack",
+                nuspec,
+                "-Verbosity", "detailed",
+                "-OutputDirectory", outputDir,
+                "-Version", version,
+                "-Properties", props);
+        }
+
+        private IEnumerable<Tuple<string, string>> GetDependencies(ProjectContext context)
+        {
+            // copied from https://github.com/dotnet/cli/blob/rel/1.0.0/src/dotnet/commands/dotnet-pack/PackageGenerator.cs
+            var project = context.RootProject;
+
+            foreach (var dependency in project.Dependencies)
+            {
+                if (dependency.Type.Equals(LibraryDependencyType.Build))
+                {
+                    continue;
+                }
+
+                // TODO: Efficiency
+                var dependencyDescription = context.LibraryManager.GetLibraries().First(l => l.RequestedRanges.Contains(dependency));
+
+                // REVIEW: Can we get this far with unresolved dependencies
+                if (dependencyDescription == null || !dependencyDescription.Resolved)
```

Comment:
`dependencyDescription` won't be `null`. if the first element can't be found, exception will be throw from previous `First` clause.